qa/lucene-index-compatibility/src/javaRestTest/java/org/elasticsearch/lucene/RollingUpgradeLuceneIndexCompatibilityTestCase.java

```diff
@@ -181,6 +182,76 @@ public void testIndexUpgrade() throws Exception {
         }
     }
 
+    /**
+     * Creates an index on N-2, closes it on N-1 and then upgrades the cluster.
+     */
+    public void testClosedIndexUpgrade() throws Exception {
+        final String index = suffix("closed-rolling-upgraded");
+        final int numDocs = 1543;
+
+        if (isFullyUpgradedTo(VERSION_MINUS_2)) {
+            createIndex(
+                client(),
+                index,
+                Settings.builder()
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .put(IndexSettings.INDEX_SOFT_DELETES_SETTING.getKey(), true)
```

Comment:
I see this in other places in this class so is probably necessary - though I do not understand why? Does not seem to be randomized in `createIndex`?

Reply:
No it's not necessary at all. And I've looked at those tests for days...  :facepalm: